### PR TITLE
Add pipeline status configuration to PNNL

### PR DESCRIPTION
**Pull request by cameronrutherford**
cc @ryandanehy

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,4 +10,11 @@ repos:
...
-    - id: autopep8
\ No newline at end of file
+    - id: autopep8
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    - id: check-toml
+    - id: forbid-new-submodules
+    - id: end-of-file-fixer
+    - id: check-yaml
```
Comment (cameronrutherford) [inline .pre-commit-config.yaml:20]: These changes were made and applied to some files - I can undo this, but I think these checks are reasonable

**Comment by cameronrutherford** [comment]
Still have to:
- [ ] Fix Spack module pipelines with new secrets/repo target
- [ ] Test ways to skip CI and not do a push mirror/only trigger certain pipelines
- [ ] Add Crusher CI pipeline status
  - [ ] Extend spack pipelines to Ascent and Crusher
- [ ] Add CPU only build in GitHub actions

**Comment by cameronrutherford** [comment]
Incline partition and Ascent success/failure should be fixed before we merge this but it's looking good!

**Comment by jaelynlitz** [comment]
> Still have to:
> 
>     * [ ]  Fix Spack module pipelines with new secrets/repo target
> 
>     * [ ]  Test ways to skip CI and not do a push mirror/only trigger certain pipelines
> 
>     * [ ]  Add Crusher CI pipeline status
>       
>       * [ ]  Extend spack pipelines to Ascent and Crusher
> 
>     * [ ]  Add CPU only build in GitHub actions
> 
> 
> For future PRs though. I am happy with these changes as is (assuming CI passes)

Did you put these in an issue? 😉

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jaelynlitz) [reply .pre-commit-config.yaml:20]: Seems reasonable to me

File: datafiles/10_scenarios_ACTIVSg10k_mod.csv [hunk 2]
```diff
@@ -8,4 +8,4 @@ scenario_nr,10691_Wind_1,10699_Wind_1,10703_Wind_1,10746_Wind_1,10768_Wind_1,107
...
...69999987,11.6769858,39.70000148,16.50000066,29.58962107,61.50000095,31.00000024,19.99857419,48.12403277,48.96607044,40.41587143,9.967713386,203.5000086,101.9999981,143.9459375,18.84099339,1.636179328,98.09193272,102.4000049,3.572407808,3.516091369,4.551286147,42.83283759,0.697925025,10.01579416,4.561259367,74.99834949,28.99999917,69.57103647,7.302727912,162.0000005,7.83615869,3.754458189,3.876564145,0.067723029,3.269732229,2.657886952,2.16552788,2.517375648,12.54317081,0.6395147,1.027601864,26.42986998,6.480752362,11.50000021,16.79999977,15.94142327,26.88194474,34.55044807,59.41783506,48.39004908,1.351121269,1.181227464,19.54092664,79.93876131,61.57021356,57.51195335,9.765084624,14.83803147,23.73789657,7.832334757,7.864212036,3.362163067,7.767246008,6.683862686,7.63118124,10.49225378,4.463883877,2.825516652,11.43529987,12.46730399,14.32611799,1.886385832,17.22238473,11.90836096,7.282169253,9.104513884,3.540326804,6.839416504,2.089503527,6.560408235,8.576190114,4.481343627,9.695919037,22.4999994,11.29758835,31.96894147,14.35076523,9.522916794,9.016575813,18.42316723,0.552794839,0.016641427,0.074900277,0.817407906,0.626845002,0.604902089,0.90112859,2.388677176,3.743133508,0.902333396
```
Comment (jaelynlitz) [inline datafiles/10_scenarios_ACTIVSg10k_mod.csv:11]: what is it doing to csv files? oh just end of file fix?

**Review by jaelynlitz — DISMISSED** [review]

File: .gitlab/pnnl/base.gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+      if [[ ! -e partition ]]; then
+        echo "No partition file found"
+        export part="none"
+      else
+        export part=$(cat partition)
+      fi
+      
+      export newell_status="ppc64le/gcc@8.5/cuda@11.4/v100@70"
+      export deception_status="x86_64/gcc@9.1/cuda@11.4"
+      export incline_status="arm64/clang@15.0/rocm@5.3/MIs100@gfx908"
```
Comment (cameronrutherford) [inline .gitlab/pnnl/base.gitlab-ci.yml:332]:
```suggestion
      export incline_status="arm64/clang@15.0/rocm@5.3/MI100@gfx908"
```

**Comment by cameronrutherford** [comment]
Testing the module rebuild scripts with the latest commit.

For some reason no Ascent test status is posted if module pipelines are run. I would expect that the pipelines would still run with .pre and .post stages, however that only works on PNNL systems through virtue of our dummy cleanup job.

I am taking this as a feature, not a bug. Another consideration I had was avoiding the Ascent push mirror pipeline altogether, but I think this is good behaviour.

**Comment by cameronrutherford** [comment]
> > Still have to:
> > ```
> > * [ ]  Fix Spack module pipelines with new secrets/repo target
> > 
> > * [ ]  Test ways to skip CI and not do a push mirror/only trigger certain pipelines
> > 
> > * [ ]  Add Crusher CI pipeline status
> >   
> >   * [ ]  Extend spack pipelines to Ascent and Crusher
> > 
> > * [ ]  Add CPU only build in GitHub actions
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > For future PRs though. I am happy with these changes as is (assuming CI passes)
> 
> Did you put these in an issue? 😉

See #11 #10 and #9

**Comment by cameronrutherford** [comment]
Skipping CI doesn't work... See #12

**Comment by cameronrutherford** [comment]
Seems like there is a race condition between all the different runners whereby if more than once job tries to get the lock on `~/.gitconfig` for the svcexasgd user at the same time that all jobs fail...

cc @kfox1111 (might be good to add you to the git repo...)

**Comment by cameronrutherford** [comment]
For some reason:
- [ ] Build pipelines are failing for seemingly no reason
- [ ] Success/Failure status indicators are paused and waiting for upsteam pipelines. Need to make sure these run as soon as tests are complete and that they are independent of upstream jobs...

**Review by ryandanehy — APPROVED** [review]
Looks good. No complaints from me. Pipeline passed so good to go.

**Pull request merged**